### PR TITLE
fast-generic-serializer supports 'enum' and 'fixed' types

### PR DESCRIPTION
**Pull request by krisso-rtb**
PR aims to fix scenario when fast-generic-serializer serializes specific record with fields of type 'enum' and/or 'fixed'.

The 4th commit it the most important one and its message (`[fast-avro] Now fast-generic-serializer supports records with 'enum' and 'fixed' types.`) might be used after squash.
1st and 2nd commit are self-explanatory, just to avoid pollution of the main commit.
Other commits were pushed just to present differences in the generated code.

----

Example stacktrace before:
```
class com.linkedin.avro.fastserde.generated.avro.JustSimpleEnum cannot be cast to class org.apache.avro.generic.GenericData$EnumSymbol (com.linkedin.avro.fastserde.generated.avro.JustSimpleEnum and org.apache.avro.generic.GenericData$EnumSymbol are in unnamed module of loader 'app')
java.lang.ClassCastException: class com.linkedin.avro.fastserde.generated.avro.JustSimpleEnum cannot be cast to class org.apache.avro.generic.GenericData$EnumSymbol (com.linkedin.avro.fastserde.generated.avro.JustSimpleEnum and org.apache.avro.generic.GenericData$EnumSymbol are in unnamed module of loader 'app')
	at com.linkedin.avro.fastserde.generated.serialization.AVRO_1_10.FastSerdeEnums_GenericSerializer_957378572.serializeFastSerdeEnums0(FastSerdeEnums_GenericSerializer_957378572.java:26)
	at com.linkedin.avro.fastserde.generated.serialization.AVRO_1_10.FastSerdeEnums_GenericSerializer_957378572.serialize(FastSerdeEnums_GenericSerializer_957378572.java:19)
	at com.linkedin.avro.fastserde.generated.serialization.AVRO_1_10.FastSerdeEnums_GenericSerializer_957378572.serialize(FastSerdeEnums_GenericSerializer_957378572.java:11)
	at com.linkedin.avro.fastserde.FastGenericSerializerGeneratorTest.dataAsBinaryDecoder(FastGenericSerializerGeneratorTest.java:615)
```

### CODE REVIEW

File: fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java [hunk 1]
```diff
@@ -296,12 +293,12 @@ private void processUnion(final Schema unionSchema, JExpression unionExpr, JBloc
...
     for (Schema schemaOption : unionSchema.getTypes()) {
       if (Schema.Type.NULL.equals(schemaOption.getType())) {
-        /**
+        /*
          * We always handle the null branch of the union first, otherwise, it leads to a bug in the
          * case where there is an optional field where the null is the second branch of the union.
          */
         JExpression condition = unionExpr.eq(JExpr._null());
-        ifBlock = ifBlock != null ? ifBlock._elseif(condition) : body._if(condition);
+        ifBlock = body._if(condition);
```
Comment (FelixGV) [inline fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java:303]: What is the rationale for this change?

**Review by FelixGV — COMMENTED** [review]
Hi, thanks a lot for this contribution!

The main change LGTM, but I have a concern about one of the "cleanups" in the first comment... can you elaborate on that one?

File: fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java (hunk 1 above)
Comment (krisso-rtb) [reply fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java:303]: Sure. Let's consider two key facts:
- For the first time we enter this block `ifBlock` is `null`
- The block can be entered at most once due to `break` at the end.

It means condition `ifBlock != null` is always `false`.
"Always" in this context means "at most once".

Comment (FelixGV) [reply fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java:303]: I see. That's right... I missed that detail when I refactored this code in https://github.com/linkedin/avro-util/commit/608bce84afc4af19fc6d9d70b0b8be02afa477df

**Review by FelixGV — APPROVED** [review]
LGTM!
